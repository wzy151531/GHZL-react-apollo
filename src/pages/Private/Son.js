import React, { PureComponent } from 'react';

export default class Son extends PureComponent {
  render() {
    console.log('Son PureComponent render');
    return (
      <div>{this.props.sonMsg}</div>
    );
  }
}