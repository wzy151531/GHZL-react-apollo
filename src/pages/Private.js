import React from 'react';
import { Row, Col, Button } from 'antd';
import Son from './Son';

export default class Private extends React.Component {
  state = {
    parentMsg: 'parentMsg',
    sonMsg: 'sonMsg',
  };

  changeParentMsg = parentMsg => {
    this.setState({
      parentMsg,
    });
  };

  render() {
    return (
      <div>Private.
        <p>This page can only be accessed by admin.</p>
        <Row type="flex" align="middle" gutter={16}>
          <Col>{this.state.parentMsg}</Col>
          <Col><Button type="primary" onClick={() => this.changeParentMsg('fuck')}>changeParentMsg</Button></Col>
        </Row>
        <Row>
          <Son sonMsg={this.state.sonMsg}/>
        </Row>
      </div>
    );
  }
};