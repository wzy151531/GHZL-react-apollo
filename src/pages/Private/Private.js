import React from 'react';
import { Row, Col, Button } from 'antd';
import Son from './Son';

export default class Private extends React.Component {
  state = {
    parentMsg: 'parentMsg',
    sonMsg: 'sonMsg',
  };

  changeParentMsg = parentMsg => {
    this.setState({
      parentMsg,
    });
  };

  showSth = () => {
    console.log('showSth');
    return 1;
  };

  render() {
    return (
      <div>Private.
        <p>This page can only be accessed by admin.</p>
        <Row type="flex" align="middle" gutter={16}>
          <Col>{this.state.parentMsg}</Col>
          <Col><Button id="changeParentMsg" type="primary" onClick={() => this.changeParentMsg('fuck')}>changeParentMsg</Button></Col>
        </Row>
        <Row>
          <Son sonMsg={this.state.sonMsg}/>
        </Row>
      </div>
    );
  }
};