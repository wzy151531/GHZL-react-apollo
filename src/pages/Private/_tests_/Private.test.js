import React from 'react';
import Private from '../Private';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

test('Private should render correctly', () => {
  const component = renderer.create(
    <Private/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should change text parentMsg state with button click', () => {
  const component = mount(<Private/>);
  expect(component.state('parentMsg')).toEqual('parentMsg');
  expect(component.state('sonMsg')).toEqual('sonMsg');

  component.find('button#changeParentMsg').simulate('click');
  // component.instance().changeParentMsg('fuck');
  expect(component.state('parentMsg')).toEqual('fuck');
  component.unmount();
});

test('callMethod of component', () => {
  const component = shallow(<Private/>);
  const result = component.instance().showSth();
  console.log(result);
});