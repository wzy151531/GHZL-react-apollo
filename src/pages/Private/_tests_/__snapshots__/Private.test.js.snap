// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private should render correctly 1`] = `
<div>
  Private.
  <p>
    This page can only be accessed by admin.
  </p>
  <div
    className="ant-row-flex ant-row-flex-middle"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      parentMsg
    </div>
    <div
      className="ant-col"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <button
        className="ant-btn ant-btn-primary"
        id="changeParentMsg"
        onClick={[Function]}
        type="button"
      >
        <span>
          changeParentMsg
        </span>
      </button>
    </div>
  </div>
  <div
    className="ant-row"
  >
    <div>
      sonMsg
    </div>
  </div>
</div>
`;
